steam/ext/dota2/models.py

```diff
@@ -0,0 +1,490 @@
+            "include_event_games": include_event_games,
+        }
+        if start_at_match_id:
+            kwargs["start_at_match_id"] = start_at_match_id
+        if hero:
+            kwargs["hero_id"] = hero.id
+
+        proto = await self._state.fetch_match_history(**kwargs)
+        return [MatchHistoryMatch(self._state, match) for match in proto.matches]
+
+
+class User(PartialUser, user.User):  # type: ignore
+    __slots__ = ()
+
+
+class PartialMatch:
+    """Represents an already finished Dota 2 Match.
+
+    This class allows using Dota 2 Coordinator requests related to matches.
+    """
+
+    def __init__(self, state: GCState, id: int):
+        self._state = state
+        self.id = id
+
+    async def details(self) -> MatchDetails:
+        """Fetch Match Details.
+
+        Contains most of the information that can be found in post-match stats in-game.
+
+        Raises
+        ------
+        asyncio.TimeoutError
+            Request time-outed. Potential reasons:
+                * Match ID is incorrect.
+                * This match is still live.
+                * Dota 2 Game Coordinator lagging or being down.
+        """
+        proto = await self._state.fetch_match_details(match_id=self.id)
+        return MatchDetails(self._state, proto.match)
+
+    async def minimal(self) -> MatchMinimal:
+        """Fetches basic "minimal" information about the match."""
+        proto = await self._state.fetch_matches_minimal(match_ids=[self.id])
+        match = next(iter(proto.matches), None)
+        if match is not None:
+            return MatchMinimal(self._state, match)
+        else:
+            msg = f"Failed to get match_minimal for {self.id}"
+            raise ValueError(msg)
+
+
+class ClientUser(PartialUser, ClientUser_):  # type: ignore
+    # TODO: if TYPE_CHECKING: for inventory
+
+    async def glicko_rating(self) -> GlickoRating:
+        """Request Glicko Rank Information."""
+        proto = await self._state.fetch_rank(rank_type=ERankType.RankedGlicko)
+        return GlickoRating(
+            mmr=proto.rank_value, deviation=proto.rank_data1, volatility=proto.rank_data2, const=proto.rank_data3
+        )
+
+    async def behavior_summary(self) -> BehaviorSummary:
+        """Request Behavior Summary."""
+        proto = await self._state.fetch_rank(rank_type=ERankType.BehaviorPublic)
+        return BehaviorSummary(behavior_score=proto.rank_value, communication_score=proto.rank_data1)
+
+    async def post_social_message(self, message: str) -> None:
+        """Post message in social feed.
+
+        Currently, messages sent with this are visible in "User Feed - Widget" of Profile Showcase.
+        This functionality was possible long ago naturally in the Dota 2 client.
+        """
+        await self._state.post_social_message(message=message)
+
+
+class ProfileCard:
+    def __init__(self, proto: common.ProfileCard):
+        self.account_id = proto.account_id
```

Comment:
This should have a user attribute rather than account id which should be at least a partial user